File: src/main/java/com/rs/java/game/player/actions/skills/crafting/leather/ReqItem.java
package com.rs.java.game.player.actions.skills.crafting.leather;

import com.rs.kotlin.Rscm;

public final class ReqItem {

    private final Object id;
    private final int amount;

    public ReqItem(Object id, int amount) {
        this.id = id;
        this.amount = amount;
    }

    public int getId() {
        if (id instanceof Integer)
            return (Integer) id;
        return Rscm.lookup((String) id);
    }

    public static ReqItem item(Object id, int amount) {
        return new ReqItem(id, amount);
    }

    public static ReqItem[] requiredItems(ReqItem... items) {
        return items;
    }


    public int getAmount() {
        return amount;
    }
}

File: src/main/java/com/rs/java/game/player/actions/skills/crafting/ReqItem.kt
package com.rs.java.game.player.actions.skills.crafting

import com.rs.kotlin.rscm.Rscm

class ReqItem(
    private val idRef: Any,
    val amount: Int,
) {
    fun getId(): Int = idRef as? Int ?: Rscm.lookup(idRef as String)

    companion object {
        @JvmStatic
        fun item(
            id: Any,
            amount: Int,
        ): ReqItem = ReqItem(id, amount)

        @JvmStatic
        fun requiredItems(vararg items: ReqItem): Array<ReqItem> = arrayOf(*items)
    }
}
